HELP.md

```diff
@@ -0,0 +1,761 @@
+contain more elements than the amount of gas available to read them all.
+In order to expose them all through view calls, we can implement pagination.
+
+`Vector` returns elements by index natively using `.get(index)`.
+
+To access elements by index in `UnorderedSet` we can use `.as_vector()` that will return a `Vector` of elements.
+
+For `UnorderedMap` we need to get keys and values as `Vector` collections, using `.keys_as_vector()` and `.values_as_vector()` respectively.
+
+Example of pagination for `UnorderedMap`:
+
+```rust
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]
+pub struct Contract {
+    pub status_updates: UnorderedMap<AccountId, String>,
+}
+
+#[near_bindgen]
+impl Contract {
+    /// Retrieves multiple elements from the `UnorderedMap`.
+    /// - `from_index` is the index to start from.
+    /// - `limit` is the maximum number of elements to return.
+    pub fn get_updates(&self, from_index: u64, limit: u64) -> Vec<(AccountId, String)> {
+        let keys = self.status_updates.keys_as_vector();
+        let values = self.status_updates.values_as_vector();
+        (from_index..std::cmp::min(from_index + limit, self.status_updates.len()))
+            .map(|index| (keys.get(index).unwrap(), values.get(index).unwrap()))
+            .collect()
+    }
+}
+```
+
+## `LookupMap` vs `UnorderedMap`
+
+### Functionality
+
+- `UnorderedMap` supports iteration over keys and values, and also supports pagination. Internally, it has the following structures:
+  - a map from a key to an index
+  - a vector of keys
+  - a vector of values
+- `LookupMap` only has a map from a key to a value. Without a vector of keys, it doesn't have the ability to iterate over keys.
+
+### Performance
+
+`LookupMap` has a better performance and stores less data comparing to the `UnorderedMap`.
+
+- `UnorderedMap` requires `2` storage reads to get the value and `3` storage writes to insert a new entry.
+- `LookupMap` requires only one storage read to get the value and only one storage write to store it.
+
+### Storage space
+
+`UnorderedMap` requires more storage for an entry comparing to a `LookupMap`.
+
+- `UnorderedMap` stores the key twice (once in the first map and once in the vector of keys) and value once. It also has higher constant for storing the length of vectors and prefixes.
+- `LookupMap` stores key and value once.
+
+## `LazyOption`
+
+It's a type of persistent collection that only stores a single value.
+The goal is prevent contract from deserializing the given value until it's needed.
+An example can be a large blob of metadata that is only needed when it's requested in a view call,
+but not needed for the majority of contract operations.
+
+It acts like an `Option` that can either hold a value or not and also requires a unique prefix (a key in this case)
+like other persistent collections.
+
+Comparing to other collections, the `LazyOption` allows to initialize the value during the constructor.
```

Comment:
Mixing grammar and substantive changes with this suggestion; not sure if I understand the substantive parts enough to convey them accurately

```suggestion
Compared to other collections, `LazyOption` only allows you to initialize the value during initialization.
```